src/server/.tests/mocha/routes/controllers/email.spec.js

```diff
@@ -0,0 +1,192 @@
+const emailController = require('../../../../routes/controllers/email')
+const assert = require('chai').assert
+const sinon = require('sinon')
+const crypto = require('crypto')
+const nodemailer = require('nodemailer')
+const DUMMY_EMAIL = 'dummy@mail.com'
+const DUMMY_SUBJECT = 'subject'
+const DUMMY_BODY = 'bodybodybody'
+const DUMMY_USER = 'dummyUser'
+const DUMMY_PASSWORD = 'dummyPassword'
+const DUMMY_HOST = 'localhost'
+const DUMMY_TOKEN = '1234567890!#'
+const NO_EMAIL = [null, null, null]
+const EXPECTED_ERROR_MESSAGE = 'Mail options cannot be undefined.'
+const BUFFER_ALLOC_SIZE = 20
+const INVALID_TOKEN = null
+const INVALID_USER = null
+const DUMMY_OPTIONS = {
+    to: DUMMY_EMAIL,
+    subject: DUMMY_SUBJECT,
+    text: DUMMY_BODY
+}
+const BUFFER_VALUE = 2
+const HEX = 'hex'
+
+const RES_MOCK = {
+    _code: "",
+    json: (obj) => {
+        return {
+            CODE: _code,
+            DESCRIPTION: obj.DESCRIPTION
+        }
+    },
+    status: function(code){
+        _code = code
+        return this
+    }
+}
+
+describe('E-mail Controller', () => {
+    describe('Sandboxed Token Creation', () => {
+        let sandbox
+
+        beforeEach(() => {
+            sandbox = sinon.sandbox.create()
+        })
+
+        afterEach(() => {
+            sandbox.restore()
+        })
+        it('Should create a valid token of length 20', () => {
+            const buf1 = Buffer.alloc(BUFFER_ALLOC_SIZE, BUFFER_VALUE)
+            sandbox.stub(crypto, 'randomBytes').returns(buf1)
+
+            const generatedToken = emailController.createToken()
+            const EXPECTED_TOKEN = buf1.toString(HEX)
+            const EXPECTED_LENGTH = 40
+
+            assert.strictEqual(generatedToken, EXPECTED_TOKEN)
+            assert.strictEqual(generatedToken.length, EXPECTED_LENGTH)
+        })
+        it('Should return a different value every time', () => {
+            const generatedToken1 = emailController.createToken()
+            const generatedToken2 = emailController.createToken()
+            assert.notStrictEqual(generatedToken1, generatedToken2)
+        })
+    })
+    describe('Mail Options', () => {
+      it('Should be equal.', () => {
+        const generatedOptions = emailController.mailOptions(DUMMY_EMAIL, DUMMY_SUBJECT, DUMMY_BODY)
+        const EXPECTED_OPTIONS = {
+            from: '"Lokim Messenger Services" <lokim.messenger@mail.com>',
```

Comment:
Extract `Lokim Messenger Services` and `lokim.messenger@mail.com` to consts at the top of the file.